packages/circuit-ui/components/ImageInput/ImageInput.tsx

```diff
@@ -293,21 +326,71 @@ export const ImageInput = ({
     clearInputElement();
   };
 
+  const handlePaste = (event: ClipboardEvent) => {
+    const { files } = event.clipboardData;
+    handleChange(files);
+
+    if (inputRef.current && files) {
+      // An error is thrown when trying to assign anything but a FileList here.
+      // For security reasons, it's not possible to simulate a FileList object.
+      // That's why this code has to be disabled during testing.
+      if (process.env.NODE_ENV !== 'test') {
+        inputRef.current.files = files;
+      }
+    }
+  };
+
+  const handleDragging = (event: DragEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setDragging(true);
+  };
+
+  const handleDragLeave = (event: DragEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setDragging(false);
+  };
```

Comment:
Could we make this a single method that just toggles `dragging`? Or would `onDragEnter` and `onDragOver` trigger it twice?

Reply:
`onDragOver` fires every hundred milliseconds or so.

Reply:
😱 so are we calling `handleDragging` every time? Is there a way to prevent setting state by `if (dragging) { return }`?

Reply:
Yes. There's no need for that check since React is smart enough not to rerender when the state doesn't change.